### ISSUE TITLE
Publish to already existing resource group.

### ISSUE THREAD

**Issue by Mr-PrashantShinde** [microsoft/BotFramework-Composer#1877]
**Is your feature request related to a problem? Please describe.**
Currently publishing only works by creating a new resource group in Azure.

**Describe the solution you'd like**
Functionality to deploy bot to already existing resource group.

**Describe alternatives you've considered**
I have tried changing the deploy script but was not successful.

**Comment by luhan2017** [comment]
@Mr-PrashantShinde , create.ps1 is basically provisioning the required azure services and put some settings in appsetting.json.  which includes 

- MicrosoftAppId
- MicrosoftAppPassword
- luis related settings (luisAuthoringKey, luisEndpointKey and luisAuthoringRegoin)
"luis": {
    "name": "ToDoBotWithLuisSample-0",
    "authoringRegion": "westus",
    "defaultLanguage": "en-us",
    "environment": "composer"
  }

If you want to run the deploy script directly, you need to set them in you appsettings.json.
Could you share more info about your appsettings.json in both root folder and ComposerDialogs\settings folder?

**Comment by cwhitten** [comment]
@Mr-PrashantShinde when you are able to provide the information @luhan2017 is looking for we can continue to dig and understand the issue. Thanks!

Incidentally, `deploy.ps1` is the script used when resources have already been provisioned.

**Comment by Mr-PrashantShinde** [comment]
Hi @luhan2017 ,

Apologies for the delay.

Please find below the appsettings.json.

Root appseting.json

`{
  "microsoftAppId": "665270b5-575b-4ba5-af13-f8230a2c56ab",
  "bot": "ComposerDialogs",
  "cosmosDb": {
    "authKey": "",
    "collectionId": "botstate-collection",
    "cosmosDBEndpoint": "",
    "databaseId": "botstate-db"
  },
  "applicationInsights": {
    "InstrumentationKey": ""
  },
  "blobStorage": {
    "connectionString": "",
    "container": "transcripts"
  }
}
`
appsettings.jon under ComposerDialogs\settings folder

`{
  "MicrosoftAppId": "665270b5-575b-4ba5-af13-f8230a2c56ab",
  "luis": {
    "name": "",
    "authoringRegion": "westus",
    "defaultLanguage": "en-us",
    "environment": "composer"
  },
  "qna": {
    "knowledgebaseid": "",
    "endpointkey": "",
    "hostname": ""
  }
}`

**Comment by cwhitten** [comment]
@luhan2017 assigning you

**Comment by luhan2017** [comment]
@Mr-PrashantShinde sorry for the delay... The deploy script not only deploy to ABS service, it also run bf-lu command to update your luis assets, you should at least include your luis authoringKey and endpoingKey in your appsettings.json. Here is the appsetting.json generated after I run the create.ps1 script:

![image](https://user-images.githubusercontent.com/32191031/77281888-54027c80-6d03-11ea-9d32-67c9c6a099b3.png)

**Comment by hibrenda** [comment]
@Mr-PrashantShinde do you still hit problems ?

**Comment by Mr-PrashantShinde** [comment]
Hi @luhan2017 and @hibrenda ,

Sorry for the delay.

Including my Luis authoringKey and endpoingKey in  appsettings.json resolved it.

**Comment by supratikchat** [comment]
Hello all,

I have built a Bot using Bot Framework composer combining qnamaker, luis and http rest calls to app insights.
However I am facing issues to publish the bot in existing RG.

Current step: Add a publish profile

Issue while executing the below:

![image](https://user-images.githubusercontent.com/67712191/88096993-60310600-cb4c-11ea-9d41-197b9e378526.png)

**Comment by luhan2017** [comment]
@supratikchat , The error message is that you don't have the permission to create a new RG. could you double check the permission and try it again?

**Comment by supratikchat** [comment]
Hello Lu,

As mentioned, I am already passing an existing resource group. I don't not need to creating a new RG. 
Can you please help on this?
Post this I can proceed with the final steps for deployment. 

Regards,
Supratik

**Comment by luhan2017** [comment]
@supratikchat , you don't need to do provisionComposer if you already created the services, you can follow this doc, https://docs.microsoft.com/en-us/composer/how-to-publish-bot#deploy-to-existing-azure-resources, just fill in some configurations in the publish profile, and then deploy the bot.

**Comment by supratikchat** [comment]
Thanks @luhan2017 for the reference. Let me try this out and proceed.

**Comment by Arjun-Jakka-Sridhar** [comment]
Hi @supratikchat,
did your problem resolved? Because, even I was also facing the same issue as yours. If you have come up with a solution please write me back or please mail me arjunjakka1998@gmail.com

**Comment by luhan2017** [comment]
@Arjun-Jakka-Sridhar , could you paste the issue here? I can help debug into it.

**Comment by Arjun-Jakka-Sridhar** [comment]
Hey @luhan2017 thanks for writing back

1. I have my existing web app service and resource in Azure
2. I have a bot created using Bot framework composer
3. I have every required keys and endpoints to paste it in the JSON for publishing


![image](https://user-images.githubusercontent.com/72132528/98893671-f48ef200-24c8-11eb-9096-84335d892774.png)

I followed this approach but in Bot framework it is not publishing 

![Untitled](https://user-images.githubusercontent.com/72132528/98893915-957dad00-24c9-11eb-9b1e-42b34c761f88.png)


I dont have any LUIS services for this bot so, i'am neglecting it to add


Please help me with this

**Comment by luhan2017** [comment]
> Hey @luhan2017 thanks for writing back
> 
> 1. I have my existing web app service and resource in Azure
> 2. I have a bot created using Bot framework composer
> 3. I have every required keys and endpoints to paste it in the JSON for publishing
> 
> ![image](https://user-images.githubusercontent.com/72132528/98893671-f48ef200-24c8-11eb-9096-84335d892774.png)
> 
> I followed this approach but in Bot framework it is not publishing
> 
> ![Untitled](https://user-images.githubusercontent.com/72132528/98893915-957dad00-24c9-11eb-9b1e-42b34c761f88.png)
> 
> I dont have any LUIS services for this bot so, i'am neglecting it to add
> 
> Please help me with this

Does you web service created from web app bot? could you share more information about that service? and what is the exactly error you met?

**Comment by Arjun-Jakka-Sridhar** [comment]
Ok @luhan2017 


This is my Web App Bot under some "xyz" resource group
My scope to this portal is resource group level scope as a contributor role

![Untitled](https://user-images.githubusercontent.com/72132528/98896283-a0870c00-24ce-11eb-8b10-5e9cba37d9c1.png)

after providing all the required keys in json to publish, the bot is not publishing and the status icon is still running and no progress on that

![image](https://user-images.githubusercontent.com/72132528/98896436-f065d300-24ce-11eb-9c63-d7ba53beaab7.png)

and I'am updating my Azure access token with new token every time it expires

**Comment by saiteja21** [comment]
hi @Arjun-Jakka-Sridhar and @luhan2017  

To publish an Azure Bot composer's Bot to an existing Resource Group, follow below solution -- 

Use the same provisioning steps defined in the README.md file however need to make a small change...

Change the "**const resourceGroupName = `${name}-${environment}`;**" value to "**const resourceGroupName = `${name}`;**" in the file **provisionComposer.js** 

Why this will work?
`const createResourceGroup = async (client, location, resourceGroupName) => {
  logger({
    status: BotProjectDeployLoggerType.PROVISION_INFO,
    message: `> Creating resource group ...`,
  });
  const param = {
    location: location,
  };

  return await client.resourceGroups.createOrUpdate(resourceGroupName, param);
};`

[client.resourceGroups.createOrUpdate](https://docs.microsoft.com/en-us/rest/api/resources/resourcegroups/createorupdate) - It just check if the RG exsists or not, it's present it'll use the same and also if you provide a new location it'll update the location to existing RG.

**Comment by Arjun-Jakka-Sridhar** [comment]
Hey @saiteja21 ,
Thanks for analyzing the problem.
Your solution worked...!

**Comment by Arjun-Jakka-Sridhar** [comment]
Also, @saiteja21 and @luhan2017 

I found various problems while publishing bot to Azure from Desktop Bot Composer app
So, I found this https://github.com/microsoft/BotFramework-Composer Open-Source Web app of this Composer and made it run in my local and it's now working completely fine.
Just in case If anyone found problem with Desktop app of Bot composer just download it from GitHub and run it has web app.

Thank you.
